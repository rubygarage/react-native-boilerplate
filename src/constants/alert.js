// eslint-disable-next-line import/prefer-default-export
export const ALERT_TYPES = {
  error: 'error',
  success: 'success',
  info: 'info',
};
