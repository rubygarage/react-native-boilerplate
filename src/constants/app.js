export const GLOBAL_ERROR = '_error';
export const OFFLINE_FLASH_ID = 'offline';
export const OFFLINE_FLASH_PRIORITY = 0;
export const FLASH_LIFETIME = 3000;

export const PLATFORM_TYPE = {
  android: 'android',
  ios: 'ios',
};
