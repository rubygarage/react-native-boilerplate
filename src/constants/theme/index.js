export default {
  colors: {
    black: '#000000',
    white: '#ffffff',
  },
  fonts: {
  },
  sizes: {
  },
};
