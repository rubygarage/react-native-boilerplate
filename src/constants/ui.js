// eslint-disable-next-line import/prefer-default-export
export const POPOVER_PLACEMENT = {
  bottom: 'bottom',
  auto: 'auto',
};
