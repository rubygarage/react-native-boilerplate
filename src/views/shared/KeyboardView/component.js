import React from 'react';
import PropTypes from 'prop-types';
import { KeyboardAvoidingView, Platform, TouchableWithoutFeedback, Keyboard } from 'react-native';

import { PLATFORM_TYPE } from 'constants/app';

const KeyboardView = ({ children }) => (
  <KeyboardAvoidingView
    behavior="padding"
    enabled={Platform.OS === PLATFORM_TYPE.ios}
  >
    <TouchableWithoutFeedback
      testID="touchableView"
      onPress={Keyboard.dismiss}
    >
      {children}
    </TouchableWithoutFeedback>
  </KeyboardAvoidingView>
);

KeyboardView.propTypes = {
  children: PropTypes.node.isRequired,
};

export default KeyboardView;
