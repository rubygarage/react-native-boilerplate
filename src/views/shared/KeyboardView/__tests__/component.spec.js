import React from 'react';
import { shallow } from 'enzyme';
import { View, Platform, Keyboard } from 'react-native';

import { PLATFORM_TYPE } from 'constants/app';
import KeyboardView from '../component';

jest.spyOn(Keyboard, 'dismiss');

describe('KeyboardView component', () => {
  it('renders correctly with iOS platform', () => {
    Platform.OS = PLATFORM_TYPE.ios;
    const component = shallow(
      <KeyboardView>
        <View />
      </KeyboardView>,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with Android platform', () => {
    Platform.OS = PLATFORM_TYPE.android;
    const component = shallow(
      <KeyboardView>
        <View />
      </KeyboardView>,
    );
    expect(component).toMatchSnapshot();
  });

  it('dismisses keyboard', () => {
    const component = shallow(
      <KeyboardView>
        <View />
      </KeyboardView>,
    );
    component.findWhere((node) => node.prop('testID') === 'touchableView').simulate('press');
    expect(Keyboard.dismiss).toHaveBeenCalled();
  });
});
