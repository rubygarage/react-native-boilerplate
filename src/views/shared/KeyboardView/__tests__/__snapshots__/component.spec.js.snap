// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardView component renders correctly with Android platform 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={false}
  keyboardVerticalOffset={0}
>
  <TouchableWithoutFeedback
    onPress={[MockFunction]}
    testID="touchableView"
  >
    <View />
  </TouchableWithoutFeedback>
</KeyboardAvoidingView>
`;

exports[`KeyboardView component renders correctly with iOS platform 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
>
  <TouchableWithoutFeedback
    onPress={[MockFunction]}
    testID="touchableView"
  >
    <View />
  </TouchableWithoutFeedback>
</KeyboardAvoidingView>
`;
